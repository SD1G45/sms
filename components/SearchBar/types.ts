export interface SearchBarProps {
  id?: string;
  value: string;
  onValueChange: (value: string) => void;
  className?: string;
}
