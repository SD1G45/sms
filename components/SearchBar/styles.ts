import styled from "styled-components";

export const Input = styled.input`
  padding: 8px 12px;
  border: 1px solid ${(props) => props.theme.colors.lightGray};
  border-radius: 50px;
  width: 100%;
`;

export const InputContainer = styled.div`
  width: 100%;
`;

export const Icon = styled.i``;
