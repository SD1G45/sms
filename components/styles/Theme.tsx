import React from "react";
import { ThemeProvider } from "styled-components";
import ITheme from "./types";

const screenWidth = {
  minSmall: `(min-width: ${430}px)`,
  minMedium: `(min-width: ${730}px)`,
  minLarge: `(min-width: ${1024}px)`,
  maxSmall: `(max-width: ${430}px)`,
  maxMedium: `(max-width: ${730}px)`,
  maxLarge: `(max-width: ${1024}px)`,
};

const lightTheme: ITheme = {
  colors: {
    primary: "#4881F0",
    lightGray: "#BEC8D2",
    textPrimary: "#000000",
    error: "#FF7F7F",
  },
  screenWidth: { ...screenWidth },
};

interface ThemeProps {
  children: React.ReactNode;
}

const Theme: React.FC<ThemeProps> = ({ children }) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);

export default Theme;
