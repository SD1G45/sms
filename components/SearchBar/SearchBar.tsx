import React from "react";
import { Icon } from "../Checkbox/styles";
import { Input, InputContainer } from "./styles";
import { SearchBarProps } from "./types";

const SearchBar: React.FC<SearchBarProps> = ({ className }) => {
  return (
    <InputContainer className={className}>
      {/* <Icon>hello</Icon> */}
      <Input placeholder="search..." />
    </InputContainer>
  );
};

export default SearchBar;
