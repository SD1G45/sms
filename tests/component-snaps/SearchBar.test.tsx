import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import SearchBar from "../../components/SearchBar";
import Theme from "../../components/styles/Theme";

describe("<SearchBar/>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(
      <Theme>
        <SearchBar id="search" value="test" onValueChange={() => {}}/>
      </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});