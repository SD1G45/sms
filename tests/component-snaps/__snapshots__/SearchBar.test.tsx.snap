// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> renders correctly 1`] = `
.c1 {
  padding: 8px 12px;
  border: 1px solid #BEC8D2;
  border-radius: 50px;
  width: 100%;
}

.c0 {
  width: 100%;
}

<div
  className="c0"
>
  <input
    className="c1"
    id="search"
    onChange={[Function]}
    placeholder="search..."
    value="test"
  />
</div>
`;
